### ISSUE TITLE
Single-line arrays and function calls can generate incorrect indentation errors

### ISSUE THREAD

**Issue by zdenekdrahos** [squizlabs/PHP_CodeSniffer#861]
After upgrade to phpcs `2.4` or `2.5` I get errors about indentation in arrays. Following code:
```php
    public function provideFiles()
    {
        return array(
            'long description' =>
                array(0, 'something'),
            'another long description' =>
                array(1, "something else")
        );
    }
```
returns error: **6 | Line indented incorrectly; expected at least 16 spaces, found 12**. So it should like this:
```php
    public function provideFiles()
    {
        return array(
            'long description' =>
                array(0, 'something'),
                'another long description' =>
                array(1, "something else")
        );
    }
```
I found that It's also fixed when I use short array syntax:
```php
    public function provideFiles()
    {
        return [
            'long description' =>
                [0, 'something'],
            'another long description' =>
                [1, "something else"]
        ];
    }
```
Short array syntax isn't always solution. This code returns same indentation error at line with `'service 2'`:
```php
        $services = array(
            'service 1' =>
                Mockery::mock('class 1')
                    ->shouldReceive('setFilter')->once()
                    ->shouldReceive('getNbResults')->atLeast()->once()
                    ->shouldReceive('getSlice')->once()->andReturn(array())
                    ->getMock(),
            'service 2' =>
                Mockery::mock('class 2')
                    ->shouldReceive('__invoke')->once()
                    ->getMock()
        );
```
It can be fixed only with temporary variable. Error is caused by line with `andReturn(array())` (or `andReturn([])`). If I delete this line or use temporary variable it works.
```php
        $emptyArray = array();
        $services = array(
            'service 1' =>
                Mockery::mock('class 1')
                    ->shouldReceive('setFilter')->once()
                    ->shouldReceive('getNbResults')->atLeast()->once()
                    ->shouldReceive('getSlice')->once()->andReturn($emptyArray)
                    ->getMock(),
            'service 2' =>
                Mockery::mock('class 2')
                    ->shouldReceive('__invoke')->once()
                    ->getMock()
        );
```

**Comment by aik099** [comment]
Same happens if you're using `master` branch of PHP_CodeSniffer?

**Comment by zdenekdrahos** [comment]
Yes. Same error messages in:
```
  - Installing squizlabs/php_codesniffer (dev-master 4966b54)
    Cloning 4966b545db497e038b3ba01fd21c0899be9365a9
```

**Comment by gsherwood** [comment]
The problem was showing up when a closing parenthesis had it's matching opening parenthesis on the same line. This didn't happen for short arrays because there was specific code to ignore these cases. The same code is being used for parenthesis now as well.

Thanks for reporting the bug and providing such clear code samples.
